src/lib/config.ts

```diff
@@ -1052,13 +1064,13 @@ function setDeepProperty(obj, value, target) {
 export function mergeDeep(o1, o2) {
     const r = Array.isArray(o1) ? o1.slice() : Object.create(o1)
     Object.assign(r, o1, o2)
-    if (o2 === undefined) return r
+    if (o2 === undefined) return removeEmpty(r)
     Object.keys(o1)
         .filter(
             key => typeof o1[key] === "object" && typeof o2[key] === "object",
         )
         .forEach(key => Object.assign(r[key], mergeDeep(o1[key], o2[key])))
-    return r
+    return removeEmpty(r)
```

Comment:
I think this is very dangerous. We won't be able to have empty strings anywhere anymore (e.g. settings `searchengine` to an empty string means "use the browser's default search engine, for `newtab` it means that the default Tridactyl newtab page shoould be used) and I'm willing to bet that most of Tridactyl will break if `undefined` is returned in places where empty strings used to be.

Reply:
Would you prefer a magic "ANNIHILATE" value? We'd just need a config updater to swap binds with "" to ANNHILATE.